papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,553 @@
+This design pattern makes defining the functional classes at the bottom of the hierarchy trivial.
+For example, the ``JSONDict`` is implemented by inheriting from ``JSONCollection`` and ``SyncedDict``, but requires no additional code to function.
+Similarly, the ``JSONList`` class inherits from ``JSONCollection`` and ``SyncedList``.
+
+This infrastructure is also flexible enough to accommodate general modifications to the synchronization protocol.
+A prominent example is the ``BufferedCollection``, a subclass of ``SyncedCollection`` that introduces additional synchronization primitives that enable toggling synchronization to and from the underlying resource with synchronization to and from an intermediate cache for improved performance.
+Similarly to base ``SyncedCollection`` functions, different buffering behaviors' synchronization can be implemented independently of the specific backend (or even the data structure, for any buffer that supports generic objects).
+
+Applications of Synced Collections
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The new synced collections promise to substantially simplify both feature and performance enhancements to the signac framework.
+Performance improvements in the form of Redis-based storage are already possible with synced collections, and as expected they show substantial speedups over the current JSON-based approach.
+The use of the new buffering protocol has enabled us to prototype new buffering approaches that further improve performance in buffered mode.
+At a larger scale, synced collections are a critical first step to enabling different data layouts on disk, such as the use of a single tabular index (e.g.
+a SQLite database) for much faster work on homogeneous data spaces or the use of more deeply nested directory structures where a deeper hierarchy on disk offers organizational benefits.
+
+The generality of synced collections makes them broadly useful even outside the signac framework.
+The framework makes it easy for developers to create Pythonic APIs for data structures that might otherwise require significant additional implementation overhead.
+Crucially, synced collections support nesting as a core feature, something that could be quite difficult to handle for developers of custom collection types.
+Moreover, while the framework was originally conceived to support synchronization of an in-memory data structure with a resource on disk, it can just as easily be used to synchronize with another in-memory resource.
+One powerful example of this would be the use of a synced collection to provide a Pythonic API to a collection-like data structure implemented as a C or C++ extension module that could function like a Python dictionary with suitable plumbing but lacks the standard APIs expected of such a class.
```

Comment:
> lacks the standard APIs expected of such a class.

@vyasr I'm confused, are you talking about the interface for MutableMapping? If so, should we say that? If not, what do you mean?

Reply:
I am also confused by this sentence. If it functions like a Python dictionary, what standard API is it missing?

Reply:
Could we break this sentence up into 2-3 sentences?